### PR TITLE
Ports: Miscellaneous improvements

### PR DESCRIPTION
**Pull request by circl-lastname**
Here I fix up some of my ports, one byte in someone else's, and end the sort-mageddon that was AvailablePorts.md.

### CODE REVIEW

File: Ports/AvailablePorts.md [hunk 1]
```diff
@@ -6,11 +6,23 @@ This list is also available at [ports.serenityos.net](https://ports.serenityos.n
...
+| [`SDL2_net`](SDL2_net/)                             | SDL2\_net (network add-on for SDL2)                             | 2.0.1                    | https://www.libsdl.org/projects/SDL_net/                                       |
+| [`SDL2_ttf`](SDL2_ttf/)                             | SDL2\_ttf (TrueType Font add-on for SDL2)                       | 2.0.18                   | https://www.libsdl.org/projects/SDL_ttf/                                       |
+| [`SDLPoP`](SDLPoP/)                                 | Prince of Persia game                                           |                          | https://github.com/NagyD/SDLPoP                                                |
+| [`SDL_sound`](SDL_sound/)                           | SDL_sound (Abstract soundfile decoder)                          |                          | https://github.com/icculus/SDL_sound                                           |
+| [`Super-Mario`](Super-Mario/)                       | Super-Mario Clone                                               |                          | https://github.com/Bennyhwanggggg/Super-Mario-Clone-Cpp                        |
```
Comment (timschumi) [inline Ports/AvailablePorts.md:21]: This is a weird way of ordering the ports. Can we make it truly alphabetically?

File: Ports/.hosted_defs.sh [hunk 2]
```diff
@@ -5,19 +5,17 @@ SCRIPT="$(dirname "${0}")"
...
-    export HOST_AR="${AR:=ar}"
-    export HOST_RANLIB="${RANLIB:=ranlib}"
-    export HOST_PATH="${PATH:=}"
-    export HOST_READELF="${READELF:=readelf}"
-    export HOST_OBJCOPY="${OBJCOPY:=objcopy}"
-    export HOST_STRIP="${STRIP:=strip}"
-    export HOST_PKG_CONFIG_DIR="${PKG_CONFIG_DIR:=}"
-    export HOST_PKG_CONFIG_SYSROOT_DIR="${PKG_CONFIG_SYSROOT_DIR:=}"
-    export HOST_PKG_CONFIG_LIBDIR="${PKG_CONFIG_LIBDIR:=}"
-fi
```
Comment (timschumi) [inline Ports/.hosted_defs.sh:20]: Realistically, we could also just add them to the list of exceptions.

File: Meta/lint-ports.py [hunk 3]
```diff
@@ -347,6 +347,10 @@ def run():
     from_table_set = set(from_table.keys())
     ports_set = set(ports.keys())
 
+    if list(from_table.keys()) != sorted(from_table.keys()): # Dictionaries are sorted by the order of addition, converting to a set loses this property
```
Comment (timschumi) [inline Meta/lint-ports.py:350]: This comment might as well be above the `if` instead of behind it, that also solves the line length CI issue.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Ports/AvailablePorts.md (hunk 1 above)
Comment (circl-lastname) [reply Ports/AvailablePorts.md:21]: It is the same way as the GitHub sort and Python by default, alternative sorts are available like `key=str.lower` which place uppercase ones inline.

File: Ports/.hosted_defs.sh (hunk 2 above)
Comment (circl-lastname) [reply Ports/.hosted_defs.sh:20]: Is there a case in which it is run twice?

Comment (timschumi) [reply Ports/.hosted_defs.sh:20]: Not intentionally as far as I can imagine, it's probably more of a safeguard.

File: Ports/AvailablePorts.md (hunk 1 above)
Comment (timschumi) [reply Ports/AvailablePorts.md:21]: Inline sort would be much better in my opinion.

File: Ports/.hosted_defs.sh (hunk 2 above)
Comment (circl-lastname) [reply Ports/.hosted_defs.sh:20]: So should I drop this change then? Just making sure.

Comment (timschumi) [reply Ports/.hosted_defs.sh:20]: Presumably yes.

File: Ports/AvailablePorts.md (hunk 1 above)
Comment (circl-lastname) [reply Ports/AvailablePorts.md]: Changed

**Comment by circl-lastname** [comment]
Is the error message "AvailablePorts.md is not in order, please ensure that all ports are sorted alphabetically by the first field" ok? I cold change "in order" to "in the correct order", or maybe add some stub showing what's wrong.

**Comment by timschumi** [comment]
Yeah, "in the correct order" instead of "in order" sounds good. I'd also maybe replace "by the first field" with "by their name".

**Comment by circl-lastname** [comment]
"by their name" could be confused for the second field called "Name", maybe first "column" instead of field?

File: Meta/lint-ports.py [hunk 4]
```diff
@@ -347,6 +347,12 @@ def run():
     from_table_set = set(from_table.keys())
     ports_set = set(ports.keys())
 
+    # Dictionaries are sorted by the order of addition, converting to a set loses this property
+    if list(from_table.keys()) != sorted(from_table.keys(), key=str.lower):
```
Comment (linusg) [inline Meta/lint-ports.py:351]: not sure why you''re mentioning sets here, there is no set in this

File: Meta/lint-ports.py [hunk 5]
```diff
@@ -347,6 +347,12 @@ def run():
...
+        all_good = False
+        print('AvailablePorts.md is not in the correct order, please ensure that all ports are sorted alphabetically',
+              'by their name.')
```
Comment (linusg) [inline Meta/lint-ports.py:354]:
```suggestion
        print(
            'AvailablePorts.md is not in the correct order, please ensure that all ports '
            'are sorted alphabetically by their name.'
        )
```

Let's also be helpful and print out the expected order (i.e. sorted keys), computers do it differently than a human would.

File: Meta/lint-ports.py (hunk 4 above)
Comment (circl-lastname) [reply Meta/lint-ports.py:351]: It explains why I converted to .keys() manually, I can remove it though.

File: Meta/lint-ports.py (hunk 5 above)
Comment (circl-lastname) [reply Meta/lint-ports.py:354]: Just the entire list of ports?

Comment (linusg) [reply Meta/lint-ports.py:354]: sure, all the names in expected order

**Comment by circl-lastname** [comment]
Sorry for the break, it is fixed now

**Pull request merged**